Objects/unicodeobject.c

```diff
@@ -2851,6 +2851,14 @@ unicode_fromformat_arg(_PyUnicodeWriter *writer,
             default: fmt = formats[sizemod]; break;
         }
         int issigned = (*f == 'd' || *f == 'i');
+        // Format strings for sprintf are selected from constant arrays of
+        // constant strings, and the variable used to index into the arrays
+        // is only assigned known constant values. Ignore warnings related
+        // to the format string not being a string literal.
+        #if defined(__GNUC__) || defined(__clang__)
+        #pragma GCC diagnostic push
```

Comment:
Consider using `_Py_COMP_DIAG_PUSH`/`_Py_COMP_DIAG_POP`, and adding a macro like `_Py_COMP_DIAG_IGNORE_DEPR_DECLS`, to make this easier to port to other compilers.

Reply:
@encukou created macro for ignoring format nonliterals and applied it to this block